tests/store/model_registry/test_sqlalchemy_store.py

```diff
@@ -719,6 +719,40 @@ def search_versions(filter_string):
         # search using run_id_2 should return versions 2 and 3
         self.assertEqual(set(search_versions("run_id='%s'" % run_id_2)), set([2, 3]))
 
+        # search using the IN operator should return all versions
+        self.assertEqual(
+            set(
+                search_versions(
+                    "run_id IN ('{run_id_1}','{run_id_2}')".format(
+                        run_id_1=run_id_1, run_id_2=run_id_2
+                    )
+                )
+            ),
+            set([1, 2, 3]),
+        )
+
+        # search using the IN operator with bad lists should return exceptions
+        with self.assertRaises(MlflowException) as exception_context:
+            search_versions("run_id IN (1,2,3)")
+        assert exception_context.exception.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
+        assert "expected string value or punctuation" in exception_context.exception.message
+
+        # search using the IN operator with empty lists should return exceptions
+        with self.assertRaises(MlflowException) as exception_context:
+            search_versions("run_id IN ()")
+        assert exception_context.exception.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
+        assert "expected a non-empty list of string values" in exception_context.exception.message
+
+        # search using the IN operator is not allowed with other additional filters
```

Comment:
this is just for the IN query, right? we do allow multiple conditions if the conditions are not IN queries? if so, why did we make this choice here?

Reply:
Per https://github.com/mlflow/mlflow/blob/master/mlflow/store/model_registry/sqlalchemy_store.py#L765, I believe we don't allow multiple conditions in search model versions filters both in OSS and within Databricks, so this test was to make sure that we didn't accidentally mess up the error messaging while adding support for the IN operator